crates/core/tedge_api/src/builder.rs

```diff
@@ -41,6 +47,17 @@ impl MeasurementVisitor for ThinEdgeJsonBuilder {
         }
     }
 
+    fn visit_type(&mut self, value: &str) -> Result<(), Self::Error> {
```

Comment:
We should check this type is not given inside a group or have a type per group.

Is the following okay?

```
{
  "location": {
    "latitude": 32.54,
    "longitude": -117.67,
    "altitude": 98.6,
    "type": "MyMeasurement"
  }
}
```

Is this the same as :

```
{
  "type": "MyMeasurement",
  "location": {
    "latitude": 32.54,
    "longitude": -117.67,
    "altitude": 98.6
  }
}
```

The current code doesn't make the difference, but rejects:

```
{
  "location": {
    "latitude": 32.54,
    "longitude": -117.67,
    "altitude": 98.6,
    "type": "MyMeasurement"
  },
     
   "area": {
      "breadth": 32.54,
      "depth": 117.67,
      "type": "MyMeasurement"
    }
}
```

Reply:
The implementation does not allow a `type` inside a group measurement. I think @reubenmiller can clarify this.

Reply:
I agree that https://github.com/thin-edge/thin-edge.io/issues/770 needs some clarifications.

However, it's clear that one has to support not only the ability to attach a type to a measurement but also arbitrary extra data (aka custom fragments) that are simply pushed to the cloud. So I will definitely not add a specific `measurement_type` field to a thin-edge-json message, but a generic bag of non-measurement fragments. Only the c8y mapper will give a specific meaning to these fragments and notably to the `type`.

Reply:
But  @reubenmiller told that only `type` is enough for now, we do not want to support other fragments for now. That is the reason why I just implemented just for `type`.

Reply:
> But  @reubenmiller told that only `type` is enough for now, we do not want to support other fragments for now. That is the reason why I just implemented just for `type`.

I'm okay to start simple but not in a way that will have to be completely reworked when implementing step 2. Furthermore, this type is something that makes hardly sense outside Cumulocity and should be treated at the level of thin-edge-json as some extra data attached to the measurements by with no specific meaning.